src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2920,6 +2920,12 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, bool* patch_m
     LogConfiguration::configure_stdout(LogLevel::Info, true, LOG_TAGS(class, path));
   }
 
+  // Disable CDS for exploded image
+  if (!has_jimage()) {
+    UseSharedSpaces = false;
+    RequireSharedSpaces = false;
+  }
```

Comment:
I think you should call `no_shared_spaces()` instead. That way, the VM will print an error message and exit if `-Xshare:on` is specified.

Reply:
Adding `no_shared_spaces()` here makes sense but it results in a duplicate output. I will move this check up to the caller of this method to avoid redundant logs.

Reply:
Fixed!